File: EssencyStockMovementMobileAppKotlin/app/src/main/java/com/essency/essencystockmovement/data/model/StockList.kt
package com.essency.essencystockmovement.data.model

data class StockList(
    val id: Int = 0, // ID autoincremental
    val idTraceabilityStockList: Int, // 🔹 Nuevo campo agregado
    val company: String,
    val source: String,
    val sourceLoc: String?,  // Puede ser `null`
    val destination: String,
    val destinationLoc: String?,  // Puede ser `null`
    val pallet: String?,  // Puede ser `null`
    val partNo: String,
    val rev: String,
    val lot: String,
    val qty: Int,
    val productionDate: String?,  // Puede ser `null`
    val countryOfProduction: String?,  // Puede ser `null`
    val serialNumber: String?,  // Puede ser `null`
    val date: String, // Formato: YYYY-MM-DD
    val timeStamp: String, // Formato ISO 8601 "2025-01-28T14:30:00"
    val user: String,
    val contBolNum: String
)

File: Essency.Stock.Movement.MobileApp/_Essency.Stock.Movement.Maui/Models/StockList.cs
using SQLite;

namespace Essency.Stock.Movement.Maui.Models
{
    public class StockList
    {
        [PrimaryKey, AutoIncrement]
        public int ID { get; set; }

        [NotNull]
        public long IDStock { get; set; }

        [NotNull]
        public string Company { get; set; }

        [NotNull]
        public string Source { get; set; }

        public string SoucreLoc { get; set; }

        [NotNull]
        public string Destination { get; set; }


        public string DestinationLoc { get; set; }

        [NotNull]
        public string PartNo { get; set; }

        [NotNull]
        public string Rev { get; set; }

        [NotNull]
        public string Lot { get; set; }

        [NotNull]
        public int Qty { get; set; }

        [NotNull]
        public string Date { get; set; }

        [NotNull]
        public DateTime TimeStamp { get; set; }

        [NotNull]
        public string User { get; set; }

        [NotNull]
        public string ContBolNum { get; set; }
    }
}
